jms/src/main/scala/akka/stream/alpakka/jms/scaladsl/JmsConsumer.scala

```diff
@@ -5,21 +5,22 @@
 package akka.stream.alpakka.jms.scaladsl
 
 import javax.jms._
-
 import akka.NotUsed
 import akka.stream.KillSwitch
 import akka.stream.alpakka.jms._
 import akka.stream.scaladsl.Source
 
-import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
 
 object JmsConsumer {
 
   /**
    * Scala API: Creates an [[JmsConsumer]] for [[javax.jms.Message]] instances
    */
-  def apply(settings: JmsConsumerSettings): Source[Message, KillSwitch] =
-    Source.fromGraph(new JmsConsumerStage(settings))
+  def apply(settings: JmsConsumerSettings): Source[Message, KillSwitch] = {
+    require(settings.destination.isDefined, noConsumerDestination(settings))
+    Source.fromGraph(new JmsConsumerStage(settings, settings.destination.get))
```

Comment:
Now, every time a consumer or producer is created, the destination is checked and explicitly passed to the stages.

Reply:
Hmm, `settings.destination.get` might tick off some static analysis tools as it is considered unsafe for the compiler. This is because the compiler does not check we have just called `settings.destination.isDefined` just the line above. It may be safer to use extraction (and the static analysis tools definitely like it better), as follows:

```scala
settings.destination match {
  case None => throw new IllegalArgumentException(noConsumerDestination(settings))
  case Some(dest) => Source.fromGraph(new JmsConsumerStage(settings, dest))
}
```